superset/views/filters.py

```diff
@@ -48,3 +52,31 @@ def apply(self, query: Query, value: Optional[Any]) -> Query:
                 user_model.username.ilike(like_value),
             )
         )
+
+
+class BaseFilterRelatedUsers(BaseFilter):  # pylint: disable=too-few-public-methods
+
+    """
+    Filter to apply on related users. Will exclude users in EXCLUDE_USERS_FROM_LISTS
+
+    Use in the api by adding something like:
+    ```
+    filter_rel_fields = {
+        "owners": [["id", BaseFilterRelatedUsers, lambda: []]],
+        "created_by": [["id", BaseFilterRelatedUsers, lambda: []]],
+    }
+    ```
+    """
+
+    name = lazy_gettext("username")
+    arg_name = "username"
+
+    def apply(self, query: Query, value: Optional[Any]) -> Query:
+        user_model = security_manager.user_model
+        exclude_users = (
+            security_manager.get_exclude_users_from_lists()
+            if current_app.config["EXCLUDE_USERS_FROM_LISTS"] is None
+            else current_app.config["EXCLUDE_USERS_FROM_LISTS"]
```

Comment:
I'm still not sure if we need two config values. I understand setting a hard-coded `EXCLUDE_USERS_FROM_LISTS` is easy but so is setting it in a custom security manager.

Even if we do want to keep `EXCLUDE_USERS_FROM_LISTS`, you can use it in the default `get_exclude_users_from_lists` so if users set `get_exclude_users_from_lists`, `EXCLUDE_USERS_FROM_LISTS` will be ignored. This prevents them from changing both configs and bite themselves down the road.

Reply:
Overriding the method makes it possibly more dynamic and the existing config key makes it easier to discover the functionality itself.
I understand yet another config setting, but relying on configuration hidden behind overriding a method on security manager is harder to discover and forces users to override the security manager (they may have not done it)

Reply:
I'm not sure why this needs to be "easy to discover"? If someone really want this feature, they will dig into the docs of config setup then hopefully find some mention of this in the Security Manager section? If we really want to keep this config value for ease of use, then maybe there is no value in adding a security manager solution, too, as it just adds more confusion.

Reply:
Also, I think most enterprise users would probably have to have their custom security manager anyway. I understand a config value is easier to set up and would do the work for most people; re-reploying Superset every time a new service account needs to be added also doesn't sound like a big deal since it is likely not frequent. But sooner or later, we'd need to migrate current FAB views for user profiles to React and add proper REST API for suers and roles, too. When that API is added, it'd be only natural to configure things like user types in the Edit User page.

A robust and scalable user layer is critical to Superset's success among enterprise users so it'd be prudent to be more future-proof here. Superset has long suffered from not having a built-in user and access control layer, which IMO had made the access control in Superset more complex than it needs to be and hindered the advance of features like RBAC. Current overrides on top of FAB are hacky and error-prone. Dynamic string-based permissions also feel unreliable. Not the scope of this PR, but I really think there should be a discussion on what the user and auth models in Superset would look like in the future.

I'm not opposed to merging this PR as it since it gets the job done, just thought in general this area needs a little bit more considerations.

Reply:
I agree with the majority of your points. We have some initiatives for discussing string based data access permissions on the near future.